hatchery/aaa_setup_test.go

```diff
@@ -0,0 +1,29 @@
+/*
+
+/!\ DO NOT CHANGE THIS FILE'S NAME!
+
+This is ridiculous, but we need this file to be _first_ (alphabetically) when running the tests so that `TestMain`
+runs before the unit tests.
+`TestMain` is the official way to implement global test setup and teardown, but there can only be one `TestMain`,
```

Comment:
could we do something like this: https://stackoverflow.com/a/73640698/4175042 ? relying on the alphabetical order of execution makes me nervous

Reply:
yeah i don't love it either but having to copy something to every unit test is what i'm trying to avoid. i'd like to get something like pytest's `conftest`

Reply:
i updated this - i don't like having to call it before each test but lmk if you think the updated code is better